src/core.js

```diff
@@ -230,19 +230,23 @@ function resumeProcessing( test ) {
 	// A slight delay to allow this iteration of the event loop to finish (more assertions, etc.)
 	if ( defined.setTimeout ) {
 		setTimeout( function() {
-			var current = test || config.current;
-			if ( current && ( current.semaphore > 0 || current.resumed ) ) {
+			var current;
+
+			// Ensure we are resuming the current test
+			if ( test && test === config.current ) {
+				current = test;
+			} else {
+				current = config.current;
+			}
```

Comment:
I shudder a little every time I look at this code. You're absolutely right to call it brittle.